drivers/modem/modem_cellular.c

```diff
@@ -2109,44 +2109,54 @@ MODEM_CHAT_SCRIPT_DEFINE(sqn_gm02s_periodic_chat_script,
-                                                                                                   \
-	MODEM_CELLULAR_DEFINE_USER_PIPE_DATA(                                                      \
-		inst,                                                                              \
-		user_pipe_1,                                                                       \
-		CONFIG_MODEM_CELLULAR_USER_PIPE_BUFFER_SIZES                                       \
-	);                                                                                         \
-                                                                                                   \
+/* Extract the first argument (pipe name) from a pair */
+#define MODEM_CELLULAR_GET_PIPE_NAME_ARG(arg1, ...) arg1
+
+/* Extract the second argument (DLCI address) from a pair */
+#define MODEM_CELLULAR_GET_DLCI_ADDRESS_ARG(arg1, arg2, ...) arg2
+
+/* Define user pipe data using instance and extracted pipe name */
+#define MODEM_CELLULAR_DEFINE_USER_PIPE_DATA_HELPER(_args, inst)                                   \
+	MODEM_CELLULAR_DEFINE_USER_PIPE_DATA(inst,                                                 \
+					     MODEM_CELLULAR_GET_PIPE_NAME_ARG _args,               \
+					     CONFIG_MODEM_CELLULAR_USER_PIPE_BUFFER_SIZES)
+
+/* Initialize user pipe using instance, extracted pipe name, and DLCI address */
+#define MODEM_CELLULAR_INIT_USER_PIPE_HELPER(_args, inst)                                          \
+	MODEM_CELLULAR_INIT_USER_PIPE(inst,                                                        \
+				      MODEM_CELLULAR_GET_PIPE_NAME_ARG _args,                      \
+				      MODEM_CELLULAR_GET_DLCI_ADDRESS_ARG _args)
+
+/*
+ * Define and initialize user pipes dynamically
+ * Takes an instance and pairs of (pipe name, DLCI address)
+ */
+#define MODEM_CELLULAR_DEFINE_AND_INIT_USER_PIPES(inst, ...)                                       \
+	FOR_EACH_FIXED_ARG(MODEM_CELLULAR_DEFINE_USER_PIPE_DATA_HELPER,                            \
+			   (;), inst, __VA_ARGS__);                                                \
 	MODEM_CELLULAR_DEFINE_USER_PIPES(                                                          \
 		inst,                                                                              \
-		MODEM_CELLULAR_INIT_USER_PIPE(inst, user_pipe_0, 3),                               \
-		MODEM_CELLULAR_INIT_USER_PIPE(inst, user_pipe_1, 4),                               \
-	);                                                                                         \
-                                                                                                   \
+		FOR_EACH_FIXED_ARG(MODEM_CELLULAR_INIT_USER_PIPE_HELPER,                           \
+				   (,), inst, __VA_ARGS__)                                         \
+	);
+
+/* Helper to define modem instance */
+#define MODEM_CELLULAR_DEFINE_INSTANCE(inst, power_ms, reset_ms, startup_ms, shutdown_ms, _auto,   \
```

Comment:
I just don't want to use C++ reserved words, even though we're in C.
An alternative might be `start` if you prefer not to use an underscore prefix.

Reply:
Or `autostart(s)` as the name of the struct member is named?

Reply:
Using the same name as the struct member will bring error during the `macrobatics` process 😄.
Since I want this to be the same line (due to 100 columns limit) I'll use `start` instead

By doing this, I can maintain the row count and match the argument passing between macro definition and macro usage.
```
#define MODEM_CELLULAR_DEFINE_INSTANCE(inst, power_ms, reset_ms, startup_ms, shutdown_ms, start,   \
				       init_script,                                                \
				       dial_script,                                                \
				       periodic_script)                                            \
```

```
	MODEM_CELLULAR_DEFINE_INSTANCE(inst, 1500, 100, 10000, 5000, false,                        \
				       &quectel_bg95_init_chat_script,                             \
				       &quectel_bg95_dial_chat_script,                             \
				       &quectel_bg95_periodic_chat_script)
```

Reply:
Just `force-pushed` to change `_auto` to `start`.